### ISSUE TITLE
Manage dependencies

### ISSUE THREAD

**Issue by sauloperez** [coopdevs/katuma-landing-page#10]
We need to get rid of `vendor/` and use package manager so that dependencies don't get out of date. I would start using `yarn`.

**Comment by sauloperez** [comment]
I put this on hold since we need to see how this may change with Jekyll.

**Comment by sauloperez** [comment]
Interesting approach: https://github.com/mmistakes/made-mistakes-jekyll

**Comment by lumie31** [comment]
Hello, @sauloperez Would want to work on this issue but I don't fully understand what it entails. Could you explain in clearer terms?

I looked through the codebase and I see a `vendor/` directory inside the `js/` directory and I assume you want to get rid of all the content of `vendor/` and manage them in another way? But I'm still not totally clear on this

**Comment by sauloperez** [comment]
I'm glad that you ask @lumie31 !

There's a better way to manage project dependencies and that's with a [package manager](https://en.m.wikipedia.org/wiki/Package_manager). I don't know how familiar are you with the topic but in JS there's [yarn](https://yarnpkg.com/en/) among others.

We should then list each of the dependencies present in `js/vendor/` using the particular version we currently have as explained in https://yarnpkg.com/en/docs/usage.

**Comment by lumie31** [comment]
I think I understand what you mean now @sauloperez and yes, I'm familiar with how to manage packages in JS with `npm` and `yarn`. So if I get you correctly, you want to probably just install the packages directly into the project without including them in the vendor folder like it currently looks, right?

Also, this project is `Ruby-based` and I think the equivalent to JavaScript's package manager (npm, yarn) is the `Gemfile`, so I'm thinking what is to be done would be installing the packages into the `Gemfile` ? or do you have other opinions?

Finally, since I've been developing with `docker` because of the limitations of my windows pc, can I still manage those dependencies using `docker`?

**Comment by sauloperez** [comment]
That's right, Ruby packages are managed with Bundler and it's `Gemfile` but that limits it to packages written in Ruby. 

What we need instead is to delegate the responsibility of managing JS libraries to a package manager so that upgrading them becomes straightforward and easy. The only to do that is with a package manager meant for JS. Then, to use them from the HTML we will simply import them from their new location which will be `node_modules/`.  

I googled to find an example and skimming through https://simpleit.rocks/ruby/jekyll/tutorials/how-to-add-bootstrap-4-to-jekyll-the-right-way/ it seems a good one. Note, however, that our manager of choice is yarn rather than npm.

As for the docker image, I think there are two things we need to change: have Yarn available and install JS dependencies right after running `bundle install`. I could focus on doing this while you investigate the rest.

**Comment by lumie31** [comment]
From your above response, it seems to me that if all goes as planned/specified, we'd probably have a `yarn.lock` and `package.json` as part of our current folder structure to house the new dependencies as opposed to it being inside the `vendor/` folder

Currently looking into the link you referenced and will get back to you on that. I will also await your feedback on the docker image part as mentioned.

**Comment by sauloperez** [comment]
Yep! that's exactly it. Then, upgrading these dependencies will be only a command away :metal:

**Comment by sauloperez** [comment]
@lumie31 I opened https://github.com/coopdevs/katuma-landing-page/pull/71 for the docker part. You can start a branch based on it.

**Comment by lumie31** [comment]
Hello, @sauloperez I need a little more clarification here. Do I install the exact versions of the dependencies as they appear in the `vendor/` directory or I can go ahead with the latest version or a higher version? Although I have a fear of compatibility if the dependency version differs.

Also, the dependencies present in the `vendor/` directory are: `bootstrap, html5-3.6-respond-1.1.0, i18next-1.7.2 and jquery-1.10.1` and after doing some search I saw 3 of the 4 dependencies above, although I couldn't find some versions as they appear in the `vendor/` directory. 

Another thing of note is, I didn't seem to find the `html5-3.6-respond-1.1.0` package and I'm wondering why and what role the package plays in the application

**Comment by sauloperez** [comment]
Hi @lumie31 !

> Do I install the exact versions of the dependencies as they appear in the vendor/ directory or I can go ahead with the latest version or a higher version? Although I have a fear of compatibility if the dependency version differs.

IMO it's safer but also quicker to install the same version we have now. Once this is done we can think of upgrading them one by one. If we do the two things at the same we will take too long and introduce risks.

> Also, the dependencies present in the vendor/ directory are: bootstrap, html5-3.6-respond-1.1.0, i18next-1.7.2 and jquery-1.10.1 and after doing some search I saw 3 of the 4 dependencies above, although I couldn't find some versions as they appear in the vendor/ directory.

I feared that. If we can't find a specific version, let's move to the closest we can find and pray :trollface: 

> Another thing of note is, I didn't seem to find the html5-3.6-respond-1.1.0 package and I'm wondering why and what role the package plays in the application

I checked (running `git log -p -- js/vendor/html5-3.6-respond-1.1.0.min.js`) and found out that this was introduced in the very first commit of the repo, quite long ago. Given it's only used to support Internet Explorer versions older than 9, I think we can safely get rid of it. No way we are going to support IE versions that old (or even IE at all...).

I hope this answers your questions.

Btw, how are you doing with your hacktoberfest? did you reach the 4 PRs goal the get the T-shirt?

**Comment by lumie31** [comment]
@sauloperez Phew!!! You really took time to answer all the questions in detail, I really appreciate that. Let me make some pushes and hope all goes well.

As regards hacktoberfest, I haven't raised 4 PRs yet but I think I can do so before the month comes to an end. I have a full-time job so it's been more of multi-tasking for me and also my "open-source contribution morale" this year is  kinda shaky, because I think for last year's edition, I already raised about 5 PRs before the 15th of the month. Also, I think it's how well and not how far

**Comment by sauloperez** [comment]
Well, just let us know if we can do anything to help you reach the goal and help your OS contribution morale to keep high.

Btw, perhaps we could have a videocall after the Hacktoberfest to know each other? After all you've been doing for us that's a must :heart_eyes:

**Comment by lumie31** [comment]
It would be my pleasure to have the video call and get to know you more. Thank you for everything
